packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/20-manual-video-quality-selection.mdx

```diff
@@ -0,0 +1,192 @@
+
+:::note
+Actual incoming video quality depends on a number of factors, such as the quality of the source video, and network conditions. Manual video quality selection allows you to specify your preference, while the actual resolution is automatically selected from the available resolutions to match that preference as closely as possible.
+:::
+
+In this article we'll build a UI control for manual video quality selection.
+
+## Prerequisites
+
+If you haven't already bootstrapped a video calling application (our [Video Calling Tutorial](https://getstream.io/video/sdk/react/tutorial/video-calling/) is a great place to start!), here's a very simple application that we'll use as a starting point:
+
+```jsx
+import {
+  SpeakerLayout,
+  StreamCall,
+  StreamVideo,
+  StreamVideoClient,
+} from '@stream-io/video-react-sdk';
+import '@stream-io/video-react-sdk/dist/css/styles.css';
+
+const client = new StreamVideoClient({
+  apiKey: 'REPLACE_WITH_API_KEY',
+  user: {
+    id: 'REPALCE_WITH_USER_ID',
+    name: 'Jane Doe',
+  },
+  token: 'REPLACE_WITH_TOKEN',
+});
+
+const App = () => {
+  const [call, setCall] = useState(null);
+
+  useEffect(() => {
+    const newCall = client.call('appointment', callId);
+    newCall
+      .join({ create: true })
+      .then(() => setCall(newCall))
+      .catch(() => console.error('Failed to join the call'));
+
+    return () =>
+      newCall.leave().catch(() => console.error('Failed to leave the call'));
+  }, []);
+
+  if (!call) {
+    return <>Loading...</>;
+  }
+
+  return (
+    <div className="str-video">
+      <StreamVideo client={client}>
+        <StreamCall call={call}>
+          <SpeakerLayout />
+        </StreamCall>
+      </StreamVideo>
+    </div>
+  );
+};
+```
+
+## Getting and Setting Incoming Video Settings
+
+To get the current incoming video quality settings, we will use the `useIncomingVideoQualitySettings` call state hook. Most importantly, it returns the following two values:
+
+- `enabled` - a boolean flag indicating whether incoming video is enabled.
+- `preferredResolution` - if video is enabled, an object of the shape `{ width: number; height: number }` containing the current preferred resolution of the incoming video.
+
+To modify the current setting, the following two methods are available on the Call object:
+
+- `setIncomingVideoEnabled` - enables or disables incoming video, clearing any preferred resolution.
+- `setPreferredIncomingVideoResolution` - sets the preference for the incoming video resolution, enabling video if it was previously disabled.
+
+:::note
+It's also possible to set a preferred resolution per call participant. There's an optional second parameter to the `setPreferredIncomingVideoResolution`, accepting an array of participant session ids. However, in this cookbook we assume that the preferences apply to all call participants.
+:::
```

Comment:
Let's add one code snippet that explains how this can be done. Something like this:
```ts
const { useParticipants } = useCallStateHooks();
const participants = useParticipants();
const [dominantParticipant, ...otherParticipants] = participants;

call.setPreferredIncomingVideoResolution(
  incomingVideoSettingToResolution('1080p'),
  [dominantParticipant.sessionId],
);
```

Reply:
https://github.com/GetStream/stream-video-js/pull/1486/commits/63a2a92d3db1b297d0131c658557c7ec97d4eae0